app/(tabs)/index.jsx

```diff
@@ -0,0 +1,309 @@
+                            iconName = "play-circle-fill"
+                            iconSize = {24}
+                            iconColor = "white"
+                            libName = {"MaterialIcons"}
+                          />
+                          {/* <CustomIconButton
+                            handlePress = {() => {}}
+                            containerStyles = "my-3 px-3"
+                            iconName = "add-box"
+                            iconSize = {24}
+                            iconColor = "white"
+                            libName = {"MaterialIcons"}
+                          /> */}
+                          <CustomIconButton
+                            handlePress = {() => {
+                              const { updateState, refreshAudioFiles } = this.context;
+                              refreshAudioFiles();
+                              updateState();
+                            }}
+                            containerStyles = "my-3 px-3"
+                            iconName = "refresh"
+                            iconSize = {24}
+                            iconColor = "white"
+                            libName = {"MaterialIcons"}
+                          />                       
+                        </View>
+                      </View>
+                      <View className= "flex-1 mt-2 border-t-2 border-solid border-secondary">
+                        <RecyclerListView
+                          dataProvider={dataProvider} 
+                          layoutProvider={this.layoutProvider} 
+                          rowRenderer={this.rowRenderer}
+                          extendedState={{isPlaying}}
+                        />
+                        
+                      </View>
+                    </View>
+                    <View className="self-center w-full border-t-2 border-solid border-secondary">
+                      <View className = "w-[96%] self-center">
+                        <MicroPlayer
+                          isFirstLaunch = {() => {this.context.firstLaunch}}
+                          title={this.context.currentAudio.filename || "Hey, wanna listen "} 
+                          duration = {this.context.currentAudio.duration || "to some tunes?_"}
+                          onAudioPress={() => router.push('../(seps)/player')}
+                          menuPress = {() => {}}
+                          isPlaying={this.context.isPlaying}
+                          activeMusicCard={() => {}}
+                          prevBtn={this.handlePrevious}
+                          nextBtn={this.handleNext}
+                          pauseBtn={() => this.handleAudioPress(this.context.currentAudio)}
+                        />
+                      </View>
+                    </View>
+                </SafeAreaView>
+                )
+            }}
+
+          </AudioContext.Consumer>
+    );
+  }
+}
+
+export default Music
```

Comment:
its interesting, why you choose class component instead of FC

Reply:
For some reason, RecyclerListView does not play well with functional components.